### PR TITLE
txn: Keep pessimistic lock's TTL if it's greater than prewrite… (#6056)

### PR DESCRIPTION
**Pull request by MyonKeminta**
###  What have you changed?

cherry-pick #5737 to release-3.1

If a pessimistic prewrite request arrives and it carries a smaller TTL than that already wrote in the pessimistic lock, keep the pessimistic lock's TTL rather than use the TTL in the reqeust.

for more detail, please check #5737.

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

Yes

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by sticnarf** [comment]
Fix the DCO please.

**Comment by MyonKeminta** [comment]
@AndreMouche  PTAL

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -98,13 +98,14 @@ impl<S: Snapshot> MvccTxn<S> {
         lock_type: LockType,
         primary: Vec<u8>,
         short_value: Option<Value>,
+        lock_ttl: u64,
```
Comment (AndreMouche) [inline src/storage/mvcc/txn.rs]: so many arguments for this function, I think it can be refactored by @nrc ?

Comment (nrc) [reply src/storage/mvcc/txn.rs]: :+1: on master rather than here, right?

Comment (AndreMouche) [reply src/storage/mvcc/txn.rs]: I think on master is enough

**Pull request merged**